src/actions/breakpoints.js

```diff
@@ -83,6 +144,41 @@ export function enableBreakpoint(location: Location) {
 }
 
 /**
+ * Syncing a breakpoint add breakpoint information that is stored, and
+ * contact the server for more data.
+ *
+ * @memberof actions/breakpoints
+ * @static
+ * @param {String} $1.sourceId String  value
+ * @param {PendingBreakpoint} $1.location PendingBreakpoint  value
+ */
+export function syncBreakpoint(
+  sourceId: string,
+  pendingBreakpoint: PendingBreakpoint
+) {
+  return ({ dispatch, getState, client, sourceMaps }: ThunkArgs) => {
+    const { line, sourceUrl, column } = pendingBreakpoint.location;
+    const location = { sourceId, sourceUrl, line, column };
+    const breakpoint = _createBreakpoint(location, pendingBreakpoint);
+
+    const promise = syncClientBreakpoint(
+      sourceId,
+      client,
+      sourceMaps,
+      pendingBreakpoint
+    );
+
+    const action = {
+      type: "SYNC_BREAKPOINT",
+      breakpoint,
+      [PROMISE]: promise
+    };
+
+    return dispatch(action);
```

Comment:
nit: now that we've simplified this I think we can inline everything 

`dispatch({ type: SYNC_BREAKPOINT, [PROMISE]: syncClientBreakpoint() })`